src/hooks/usePagination.js

```diff
@@ -1,14 +1,28 @@
 export const DOTS = "...";
 
-function usePagination() {
+function usePagination({ currentPage, totalCount, pageSize, maxPageNumber }) {
```

Comment:
The value returned by this hook can be memoized/cached to avoid unnecessarily running the logic every time the hook is called. The useMemo() hook seems suitable.